packages/playwright-core/src/server/page.ts

```diff
@@ -809,11 +809,14 @@ export class Page extends SdkObject {
     this._isServerSideOnly = true;
   }
 
-  async snapshotForAI(metadata: CallMetadata): Promise<string> {
-    const frameIds: string[] = [];
-    const snapshot = await snapshotFrameForAI(this.mainFrame(), 0, frameIds);
-    this.lastSnapshotFrameIds = frameIds;
-    return snapshot.join('\n');
+  async snapshotForAI(options: types.TimeoutOptions, metadata: CallMetadata): Promise<string> {
+    const controller = new ProgressController(metadata, this);
+    return controller.run(async progress => {
```

Comment:
- Let's separate this into two PRs, where the first one adds a timeout?
- I'd say it would be great to return whatever frames we were able to grab during the allocated timeout, and only throw if the main frame snapshot did timeout. WDYT?

Reply:
yeah let's separate it. it turns out we don't even retry capturing a frame at the moment, we just error out - fixing that in https://github.com/microsoft/playwright/pull/35951 for starters